### ISSUE TITLE
Error in one of UD tasks while building with Ninja

### ISSUE THREAD

**Issue by rolavick** [AliceO2Group/O2Physics#4573]
Hi,

I could not build O2Physics with ninja getting this error

```
alice/sw/SOURCES/O2Physics/master/0/PWGUD/Tasks/upcTauTau13topo.cxx:278:18: error: 'auto' not allowed in function prototype
  int TrackCheck(auto track)
```

This is related to this task by @amatyja 

https://github.com/AliceO2Group/O2Physics/blob/001ca05b9742c8fe7e5ac4a7979081884e87bd96/PWGUD/Tasks/upcTauTau13topo.cxx#L278

what worked for me was to to replace 

`  int TrackCheck(auto track)`
with
```
  template <typename T>
  int TrackCheck(T track)
```

**Comment by amatyja** [comment]
Hi @rolavick , It is better to change it to proposed way.

**Comment by rolavick** [comment]
Issue closed as it was solved with #4574
